In [1]:
import pandas as pd
from scipy import stats
import numpy as np

1.     A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


In [3]:
cutlets=pd.read_csv('Cutlets.csv')
cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

 Null hypothesis Ho = there is no difference in diameter
    Alternative hypothesis ,    Ha = there is difference in diameter

In [6]:
stats.ttest_ind(cutlets['Unit A'],cutlets['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

statistic=0.723 ,p-value=0.472
Therefore , p-value>α    that is 0.472>0.05
Therefore we fail to reject null hypothesis. And there is no difference in diameter.

2.   A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.


In [7]:
labtat=pd.read_csv('LabTaT.csv')
labtat.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

Null hypothesis Ho = There is no difference in average TAT
Alternative hypothesis , Ha = There is difference between average TAT. 


In [12]:
stats.f_oneway(labtat.iloc[0:,0],labtat.iloc[0:,1],labtat.iloc[0:,2],labtat.iloc[0:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Pvalue<0.05 so we fail to accept null hypothesis.


3.Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


In [14]:
BR=pd.read_csv('BuyerRatio (1).csv')
BR.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Null Hypothesis Ho=All proportions are equal.
Alternative hypothesis,Ha=Not all proportions are equal.

In [15]:
brtable=[[50,142,131,70],[435,1523,1356,750]]

In [19]:
from scipy.stats import chi2_contingency

In [30]:
chi2_contingency(brtable)


(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [40]:
data={'Observed values':['Males','Females'],'East':[42.76531,442.2347],'West':[146.8129,1518.187],'North':[131.1176,1355.882],'South':[72.30424,747.6958]}

In [49]:
Expectedfrequencytable=pd.DataFrame(data)
Expectedfrequencytable

Observed values       East       West      North      South
0           Males   42.76531   146.8129   131.1176   72.30424
1         Females  442.23470  1518.1870  1355.8820  747.69580

chi-square=∑(𝑂𝑖−𝐸𝑖)2/𝐸𝑖

Where , Oi=observed frequency in the data for each type

   Ei=Expected frequency to see of each type
        
Chi-Square=1.22+0.158+0.0001+0.073+0.118+0.015+0.00001+0.0071

Chi-Square(actual value)=1.5959

Degrees of freedom=(number of rows-1)*(number of columns-1) =(2-1)*(4-1)=3

Chi-Square table(α=0.05 & df=3)

Critical value=7.815   

Therefore, Actual value<critical value   or   pvalue<α
                     1.6<7.815
                     
Therefore we fail to reject null hypothesis(Ho )

Therefore, All proportions are equal

4.     TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [52]:
COF=pd.read_csv('Costomer+OrderForm (1).csv')
COF

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

Null hypothesis ,Ho= The defective % does not  varies by centre.


Alternative hypothesis ,Ha =The defective % varies by centre.


In [53]:
COF.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [56]:
COF.isnull().sum()

Phillippines    0
Indonesia       0
Malta           0
India           0
dtype: int64

In [59]:
contingency_tabel=[[271,267,269,280],[29,33,31,20]]
print(contingency_tabel)

[[271, 267, 269, 280], [29, 33, 31, 20]]


In [71]:
exp=stats.chi2_contingency(contingency_tabel)
exp

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Defining Expected values And Observed values

In [72]:
observed=np.array([271,267,269,280,29,33,31,20])
expected=np.array([271.75,271.75,271.75,271.75,28.25,28.25,28.25,28.25])

In [77]:
test_statistic,p_value=stats.chisquare(observed,expected,ddof=3)
test_statistic,p_value
print("Test Statistic=" , test_statistic , '\n' ,'p_value=',p_value)

Test Statistic= 3.858960685820355 
 p_value= 0.4254298144535761


p=0.42542 we fail to reject Null Hypothesis